source/codegen/metadata_mutation.py

```diff
@@ -64,7 +64,12 @@ def mark_non_proto_params(parameters):
             if mechanism == 'len':
                 if 'determine_size_from' not in size_param:
                     size_param['determine_size_from'] = []
+                    size_param['linked_params_are_optional'] = True
```

Comment:
I'm a little confused - is there a reason to set this here since we're unconditionally setting it below?

Reply:
Oh, I see, because you're using the value to make sure there's not a mix. Nevermind.